core/json/src/main/scala/net/liftweb/json/Formats.scala

```diff
@@ -230,9 +234,16 @@ case class ShortTypeHints(hints: List[Class[_]]) extends TypeHints {
 /** Use full class name as a type hint.
  */
 case class FullTypeHints(hints: List[Class[_]]) extends TypeHints {
+  private val hintsToClass: ConcurrentScalaMap[String, Class[_]] =
+    new ConcurrentHashMap[String, Class[_]]().asScala ++= hints.map(clazz => hintFor(clazz) -> clazz)
```

Comment:
What's the performance impact of using the scala wrapper around CHM? It might be nothing, but my anecdotal impression is that using that is met with a bit of skepticism by other Scalans. I could be wrong, though, in which case I would be glad to know it.